homeassistant/components/switch/__init__.py

```diff
@@ -69,37 +74,41 @@ async def async_setup(hass, config):
     return True
 
 
-async def async_setup_entry(hass, entry):
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up a config entry."""
-    return await hass.data[DOMAIN].async_setup_entry(entry)
+    return cast(bool, await hass.data[DOMAIN].async_setup_entry(entry))
 
 
-async def async_unload_entry(hass, entry):
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
-    return await hass.data[DOMAIN].async_unload_entry(entry)
+    return cast(bool, await hass.data[DOMAIN].async_unload_entry(entry))
 
 
 class SwitchEntity(ToggleEntity):
     """Base class for switch entities."""
 
+    _attr_current_power_w: float | None = None
+    _attr_is_standby: bool | None = None
+    _attr_today_energy_kwh: float | None = None
+
     @property
-    def current_power_w(self):
+    def current_power_w(self) -> float | None:
         """Return the current power usage in W."""
-        return None
+        return self._attr_current_power_w
 
     @property
-    def today_energy_kwh(self):
+    def today_energy_kwh(self) -> float | None:
         """Return the today total energy usage in kWh."""
-        return None
+        return self._attr_today_energy_kwh
 
     @property
-    def is_standby(self):
+    def is_standby(self) -> bool | None:
```

Comment:
This property is missing from the `PROP_TO_ATTR` map which means it's not working.
Should we remove it?

Reply:
I dunno? I just kept it as documented/interfaced.
It isn't used by any core integration at least.

Reply:
It's implemented by `hdmi_cec.switch`, but my point was that it won't do anything because it's not included in the state attributes.

Reply:
Let's clean that up in another PR.